### PR TITLE
RDCC-449: OpenId implementation for prd project

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-449

### Change description ###

OpenId implementation in prd

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: src/functionalTest/resources/application-functional.yaml [hunk 1]
```diff
@@ -1,16 +1,15 @@
-targetInstance: ${TEST_URL:http://rd-professional-api-aat.service.core-compute-aat.internal}
+targetInstance: ${TEST_URL:http://rd-professional-api-pr-243.service.core-compute-preview.internal}
```
Comment (AdilOoze) [inline src/functionalTest/resources/application-functional.yaml]: Was this changed to run functional test locally? Or is it code to be committed?

File: src/main/resources/application.yaml [hunk 2]
```diff
@@ -100,7 +100,7 @@ exui.role:
 prdEnumRoleType: ADMIN_ROLE,JURISD_ID
 jurisdictionIdType: ADMIN_ROLE,CCD_ROLE,SIDAM_ROLE
 
-userProfUrl: ${USER_PROFILE_URL:https://rd-user-profile-api-pr-108.service.core-compute-preview.internal}
+userProfUrl: ${USER_PROFILE_URL:https://rd-user-profile-api-pr-115.service.core-compute-preview.internal}
```
Comment (akiohmcts) [inline src/main/resources/application.yaml]: Change to api preview

File: src/functionalTest/resources/application-functional.yaml [hunk 3]
```diff
@@ -1,16 +1,15 @@
...
 
 idam.api.url: ${IDAM_URL:https://idam-api.aat.platform.hmcts.net}
 generated.user.email.pattern: ${GENERATED_USER_EMAIL_PATTERN:freg-test-user-%s@feemail.com}
-test.url: ${TEST_URL:http://rd-professional-api-aat.service.core-compute-aat.internal}
 test.user.password: ${TEST_USER_PASSWORD:Hmcts123}
-#oauth2.redirect.uri: ${OAUTH2_REDIRECT_URI:https://rd-professional-api-preview.service.core-compute-preview.internal/oauth2redirect}
 oauth2.client.id: ${OAUTH2_CLIENT_ID:rd-professional-api}
 oauth2.client.secret: ${OAUTH2_CLIENT_SECRET:cc5f2a6-9690-11e9-bc42-526af7764f64}
 idam.auth.clientId: rd-professional-api
+idam.auth.clientSecret: cc5f2a6-9690-11e9-bc42-526af7764f64
```
Comment (akiohmcts) [inline src/functionalTest/resources/application-functional.yaml]: Always use env variables. And NEVER the aat (sensitive) hard-coded defaults.

Comment (akiohmcts) [reply src/functionalTest/resources/application-functional.yaml]: Please remove

Comment (shreedhar-hmcts) [reply src/functionalTest/resources/application-functional.yaml]: For this separate JIRA has raised and will be fixed under that.

File: src/main/resources/application.yaml (hunk 2 above)
Comment (shreedhar-hmcts) [reply src/main/resources/application.yaml]: For this separate JIRA has raised and will be fixed under that.

**Pull request merged**